subiquity/common/filesystem/sizes.py

```diff
@@ -75,3 +82,34 @@ def get_efi_size(available_space):
 def get_bootfs_size(available_space):
     all_factors = (uefi_scale, bootfs_scale, rootfs_scale)
     return scale_partitions(all_factors, available_space)[1]
+
+
+def calculate_guided_resize(part_min: int, part_size: int, install_min: int,
```

Comment:
How will this API be used when a partition is followed by a gap?

Reply:
Ah, we should at least use a gap that immediately follows a resized partition.  I think that should be a follow-on PR to extend gaps a bit so that we can easily acquire a gap after a partition, if one exists.  Armed with that it should be easy to allow that to be factored in to guided resize.